toolkit/themes/shared/design-system/config.js

```diff
@@ -263,3 +263,108 @@ module.exports = {
     },
   },
 };
+
+const TOKEN_SECTIONS = {
+  "Base tokens/Color": [/^color-[^-]*$/, /^color-.*-a?\d+$/],
+  "Application tokens/Border": "border",
+  "Application tokens/Box": "box",
+  "Application tokens/Color": "color",
+  "Application tokens/Font weight": "font-weight",
+  "Application tokens/Focus outline": "focus-outline",
+  "Application tokens/Icon": "icon",
+  "Application tokens/Input/Button": "button",
+  "Application tokens/Input/Checkbox": "checkbox",
+  "Application tokens/Input/Text": "input-text",
+  "Application tokens/Link": "link",
+  "Application tokens/Text": "text",
+  "Application tokens/Size": "size",
+  "Application tokens/Spacing": ["space", "spacing"],
+  Unspecified: "",
+};
+
+function sortByName(a, b) {
+  let aParts = a.name.split("-");
+  let bParts = b.name.split("-");
+  if (aParts.length == bParts.length) {
+    // 05 > 10 when comparing strings, so handle numbers as ints.
+    for (let i = 0; i < aParts.length; i++) {
+      let aPart = aParts[i];
+      let bPart = bParts[i];
+      if (/^\d+$/.test(aPart) && /^\d+$/.test(bPart)) {
+        if (parseInt(bPart, 10) > parseInt(aPart, 10)) {
+          return -1;
+        }
+        if (parseInt(aPart, 10) > parseInt(bPart, 10)) {
+          return 1;
+        }
+      }
+      if (bPart > aPart) {
+        return -1;
+      }
+      if (aPart > bPart) {
+        return 1;
+      }
+    }
+    return 1;
+  }
+  return b.name > a.name ? -1 : 1;
+}
```

Comment:
```suggestion
function sortByName(a, b) {
  let aParts = a.name.split("-");
  let bParts = b.name.split("-");
  if (aParts.length == bParts.length) {
    for (let i = 0; i < aParts.length; i++) {
      let aPart = aParts[i];
      let bPart = bParts[i];
      if (aPart !== bPart) {
        return aPart.localeCompare(bPart, undefined, { numeric: true });
      }
    }
    return 1;
  }
  return a.name.localeCompare(b.name);
}
```

I think this could be simplified a bit by using `localeCompare` which has an [option to handle comparing numbers](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/String/localeCompare#numeric_sorting). Not 100% sure this is right, but something along these lines 😆